### ISSUE TITLE
create app with different db name fails

### ISSUE THREAD

**Issue by durandom** [ManageIQ/manageiq-pods#133]
I want to create an app with another db name than the default but this fails on the app image. Looks like `DATABASE_NAME` is not passed down to `appliance-initialize.sh` somewhere

```
❯ oc new-app --template=manageiq -p DATABASE_NAME=vmdb_honk
--> Deploying template "myproject/manageiq" to project myproject

     manageiq
     ---------
     ManageIQ appliance with persistent storage


     * With parameters:
        * Name=manageiq
        * PostgreSQL Service Name=postgresql
        * PostgreSQL User=root
        * PostgreSQL Password=bFNQFetb # generated
        * PostgreSQL Database Name=vmdb_honk
        * Application Database Region=0
        * Memcached Service Name=memcached
        * Memcached Max Memory=64
        * Memcached Max Connections=1024
        * Memcached Slab Page Size=1m
        * PostgreSQL Max Connections=100
        * PostgreSQL Shared Buffer Amount=256MB
        * Application Min CPU Requested=1000m
        * PostgreSQL Min CPU Requested=500m
        * Memcached Min CPU Requested=200m
        * Application Min RAM Requested=6144Mi
        * PostgreSQL Min RAM Requested=1024Mi
        * Memcached Min RAM Requested=64Mi
        * Application Max RAM Limit=16384Mi
        * PostgreSQL Max RAM Limit=8192Mi
        * Memcached Max RAM Limit=256Mi
        * PostgreSQL Image Name=docker.io/manageiq/manageiq-pods
        * PostgreSQL Image Tag=postgresql-latest
        * Memcached Image Name=docker.io/manageiq/manageiq-pods
        * Memcached Image Tag=memcached-latest
        * Application Image Name=docker.io/manageiq/manageiq-pods
        * Application Image Tag=app-latest
        * Application Hostname=
        * Application Replica Count=1
        * Application Init Delay=15
        * Application Volume Capacity=5Gi
        * Application Region Volume Capacity=5Gi
        * Database Volume Capacity=15Gi

--> Creating resources ...
    secret "manageiq-secrets" created
    service "manageiq" created
    route "manageiq" created
    imagestream "miq-app" created
    imagestream "miq-postgresql" created
    imagestream "miq-memcached" created
    persistentvolumeclaim "manageiq-postgresql" created
    persistentvolumeclaim "manageiq-region" created
    statefulset "manageiq" created
    service "memcached" created
    deploymentconfig "memcached" created
    service "postgresql" created
    deploymentconfig "postgresql" created
--> Success
    Run 'oc status' to view your app.
```

```
❯ oc get pods
NAME                 READY     STATUS    RESTARTS   AGE
manageiq-0           0/1       Running   0          7m
memcached-1-93pfb    1/1       Running   0          7m
postgresql-1-7bsvz   1/1       Running   0          7m

~/src/manageiq-dos master*
❯ oc rsh manageiq-0 bash -l
[root@manageiq-0 vmdb]# systemctl status
● manageiq-0
    State: degraded
     Jobs: 1 queued
   Failed: 1 units
    Since: Fri 2017-05-05 14:41:42 UTC; 6min ago
   CGroup: /
           ├─  1 /usr/sbin/init
           ├─691 bash -l
           ├─726 systemctl status
           ├─727 less
           └─system.slice
             ├─dbus.service
             │ └─77 /bin/dbus-daemon --system --address=systemd: --nofork --nopi
             ├─crond.service
             │ └─24 /usr/sbin/crond -n
             ├─evmserverd.service
             │ ├─722 /bin/bash /bin/evmserver.sh start
             │ └─724 ruby /opt/rubies/ruby-2.3.1/bin/rake evm:start
             ├─systemd-journald.service
             │ └─20 /usr/lib/systemd/systemd-journald
             ├─evm-watchdog.service
             │ ├─178 /bin/bash /bin/evm_watchdog
             │ └─179 ruby /bin/evm_watchdog.rb
[root@manageiq-0 vmdb]# systemctl
  UNIT                      LOAD   ACTIVE     SUB       JOB   DESCRIPTION
  dev-sda1.device           loaded activating tentative       /dev/sda1
  -.mount                   loaded active     mounted         /
  dev-mqueue.mount          loaded active     mounted         POSIX Message Queu
  dev-termination\x2dlog.mount loaded active     mounted         /dev/terminatio
  etc-hostname.mount        loaded active     mounted         /etc/hostname
  etc-hosts.mount           loaded active     mounted         /etc/hosts
  etc-resolv.conf.mount     loaded active     mounted         /etc/resolv.conf
  persistent.mount          loaded active     mounted         /persistent
  persistent\x2dregion.mount loaded active     mounted         /persistent-regio
  run-secrets-kubernetes.io-serviceaccount.mount loaded active     mounted
● appliance-initialize.service loaded failed     failed          Initialize Appl
  crond.service             loaded active     running         Command Scheduler
  dbus.service              loaded active     running         D-Bus System Messa
  evm-watchdog.service      loaded active     running         evm watchdog proce
  evmserverd.service        loaded activating start     start EVM server daemon
  miqtop.service            loaded active     running         "top(1) data colle
  miqvmstat.service         loaded active     running         "vmstat(1) data co
  network.service           loaded active     exited          LSB: Bring up/down
  systemd-journald.service  loaded active     running         Journal Service
  systemd-tmpfiles-setup.service loaded active     exited          Create Volati
  -.slice                   loaded active     active          Root Slice
  system.slice              loaded active     active          System Slice

[root@manageiq-0 vmdb]# systemctl -l status appliance-initialize
● appliance-initialize.service - Initialize Appliance Database
   Loaded: loaded (/usr/lib/systemd/system/appliance-initialize.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Fri 2017-05-05 14:42:01 UTC; 6min ago
  Process: 36 ExecStart=/bin/appliance-initialize.sh (code=exited, status=1/FAILURE)
 Main PID: 36 (code=exited, status=1/FAILURE)

May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: from /usr/bin/appliance_console_cli:11:in `<main>'
May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: create encryption key
May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: configuring external database
May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: Validated failed with error - FATAL:  database "vmdb_production" does not exist
May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: .
May 05 14:42:01 manageiq-0 appliance-initialize.sh[36]: ERROR: Failed to initialize appliance, please check journal or appliance_console logs at /var/www/miq/vmdb/log/appliance_console.log
May 05 14:42:01 manageiq-0 systemd[1]: appliance-initialize.service: main process exited, code=exited, status=1/FAILURE
May 05 14:42:01 manageiq-0 systemd[1]: Failed to start Initialize Appliance Database.
May 05 14:42:01 manageiq-0 systemd[1]: Unit appliance-initialize.service entered failed state.
May 05 14:42:01 manageiq-0 systemd[1]: appliance-initialize.service failed.
```

**Comment by bazulay** [comment]
@carbonin @fbladilo PTAL 
This may also explain some failures I had with the external db template.

**Comment by fbladilo** [comment]
@durandom @bazulay Let me have a look, certainly DATABASE_NAME gets passed to PG but possibly getting dropped on the MIQ pod.

**Comment by fbladilo** [comment]
@durandom Let me know if this one is Ok to close, fix via #135

**Comment by durandom** [comment]
works. thanks for fixing :)
